ci/cudf_pandas_scripts/pandas-tests/diff.sh

```diff
@@ -8,17 +8,22 @@
 
 # Hard-coded needs to match the version deduced by rapids-upload-artifacts-dir
 MAIN_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py310.main-results.json
-PR_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py310.pr-results.json
-aws s3 cp $MAIN_ARTIFACT main-results.json
+PR_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py39.pr-results.json
+
+rapids-logger "Fetching latest available results from nightly"
+aws s3api list-objects-v2 --bucket rapids-downloads --prefix "nightly/" --query "sort_by(Contents[?ends_with(Key, '.main-results.json')], &LastModified)[::-1].[Key]" --output text > s3_output.txt
+cat s3_output.txt
+read -r COMPARE_ENV < s3_output.txt
+export COMPARE_ENV
+rapids-logger "Latest available results from nightly: ${COMPARE_ENV}"
+
+aws s3 cp "s3://rapids-downloads/${COMPARE_ENV}" main-results.json
 aws s3 cp $PR_ARTIFACT pr-results.json
 
 # Compute the diff and prepare job summary:
 python -m pip install pandas tabulate
 python ci/cudf_pandas_scripts/pandas-tests/job-summary.py main-results.json pr-results.json | tee summary.txt >> "$GITHUB_STEP_SUMMARY"
 
-COMMENT=$(head -1 summary.txt)
-
+COMMENT=$(head -1 summary.txt | grep -oP '\d+/\d+ \(\d+\.\d+%\).*?(a decrease by|an increase by) \d+\.\d+%')
 echo "$COMMENT"
-
-# Magic name that the custom-job.yaml workflow reads and re-exports
-echo "job_output=${COMMENT}" >> "${GITHUB_OUTPUT}"
+echo "{\"context\":\"Pandas tests\", \"description\":\"$COMMENT\", \"state\":\"success\", \"job_name\": \"$GH_JOB_NAME\"}" > gh-status.json
```

Comment:
I prefer not having to escape everything, so:
```suggestion
jq --arg COMMENT "$COMMENT" --arg GH_JOB_NAME "$GH_JOB_NAME" -n \
  '{"context": "Pandas tests",
    "description": $COMMENT, 
    "state":"success", 
    "job_name": $GH_JOB_NAME}' \
    > gh-status.json
```